hackathon_site/hardware/test_api.py

```diff
@@ -194,3 +194,244 @@ def test_hardware_get_success(self):
+            max_per_team=1,
+            picture="/picture/location",
+        )
+
+        self.view = reverse("api:hardware:hardware-list")
+
+    def _build_filter_url(self, **kwargs):
+        return (
+            self.view + "?" + "&".join([f"{key}={val}" for key, val in kwargs.items()])
+        )
+
+    def test_user_not_logged_in(self):
+        response = self.client.get(self.view)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_search_by_name(self):
+        self._login()
+
+        url = self._build_filter_url(search="arduino")
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        data = response.json()
+        expected_data = {
+            # Your expected data
+            "id": 2,
+            "name": "arduino",
+            "model_number": "model",
+            "manufacturer": "manufacturer",
+            "datasheet": "/datasheet/location/",
+            "quantity_available": 4,
+            "notes": "notes",
+            "max_per_team": 1,
+            "picture": "http://testserver/media/picture/location",
+            "categories": [],
+            "quantity_remaining": 4,
+        }
+
+        self.assertEqual(data["results"][0], expected_data)
+
+    def test_in_stock_filter(self):
+        self._login()
+
+        # # when in_stock is true
+        url_true = self._build_filter_url(in_stock="true")
+        expected_data_true = {
+            "count": 2,
+            "next": None,
+            "previous": None,
+            "results": [
+                {
+                    "id": 2,
+                    "name": "arduino",
+                    "model_number": "model",
+                    "manufacturer": "manufacturer",
+                    "datasheet": "/datasheet/location/",
+                    "quantity_available": 4,
+                    "notes": "notes",
+                    "max_per_team": 1,
+                    "picture": "http://testserver/media/picture/location",
+                    "categories": [],
+                    "quantity_remaining": 4,
+                },
+                {
+                    "id": 3,
+                    "name": "hardware3",
+                    "model_number": "model",
+                    "manufacturer": "manufacturer",
+                    "datasheet": "/datasheet/location/",
+                    "quantity_available": 4,
+                    "notes": "notes",
+                    "max_per_team": 1,
+                    "picture": "http://testserver/media/picture/location",
+                    "categories": [],
+                    "quantity_remaining": 4,
+                },
+            ],
+        }
+        response_true = self.client.get(url_true)
+        self.assertEqual(response_true.status_code, status.HTTP_200_OK)
+        data_true = response_true.json()
+        self.assertEqual(data_true, expected_data_true)
+
+        # when in_stock is false
```

Comment:
This should be a separate test case (`test_in_stock_false`). That way if one of them fails, we get a verbose error message about why